manifests/params.pp

```diff
@@ -198,13 +194,8 @@
 
   # Will this host be a puppet agent ?
   $agent                      = true
-  $client_certname            = $::clientcert
-
-  if defined('$::puppetmaster') {
-    $puppetmaster             = $::puppetmaster
-  } else {
-    $puppetmaster             = undef
-  }
+  $client_certname            = $trusted['certname']
+  $puppetmaster               = $server_facts['servername']
```

Comment:
This is one I wonder about. `puppetmaster` is a global variable we set in Foreman's ENC, which allows you to change the puppetserver easily. Also, setting this outside of Foreman's ENC. For example, when using DNS lookups. Should we keep the old `defined` code/update it to `getvar()`?

Reply:
I ended up changing it to `getvar('puppetmaster')`, to remain compatible.